scipy/optimize/_minimize.py

```diff
@@ -600,50 +600,67 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
     if bounds is not None:
         bounds = standardize_bounds(bounds, x0, meth)
 
+        lb, ub = _split_bounds(bounds)
+        btol = 1e-8  # tolerance for bounds being considered equal. Expose?
+        i_fixed = np.abs(ub - lb) < btol
+
+        if i_fixed.any():
+            x_fixed = ((lb+ub)/2)[i_fixed]  # values of fixed variables
+            x0 = x0[~i_fixed]   # remove fixed variables from x0
+            bounds = _remove_fixed_variables_from_bounds(bounds, i_fixed)
+            fun = _remove_fixed_variables_from_function(fun, i_fixed, x_fixed)
+            # do the same for all other functions (constraints, etc.)
```

Comment:
This is where all the other user-supplied callables get wrapped, if they exist, e.g.
`jac = _remove_fixed_variables_from_function(jac, i_fixed, x_fixed)`

Reply:
Do you mean derivative-based methods? Presumably `bounds` will be `None` (so this would not run) for methods that don't support bounds.

Reply:
> I suppose I need to remove some of the entries of jac's output afterwards, but shouldn't be a problem

Yes

> Presumably bounds will be None (so this would not run) for methods that don't support bounds

Not the case. The user can specify bounds for methods that don't use them, they're just ignored (RuntimeWarning, but no exception). If that's the case we wouldn't want to eliminate x0 entries, so a check needs to be made for what type of method it is.

Reply:
I see that methods that don't support bounds still run even if you supply bounds.  Without this check, the variables with lb==ub would be fixed but the rest of the bounds would not be respected, so yes, I added the check. Didn't create the all-caps variable yet; we need to discuss which methods this is for.